builders/tutorials/subsquid.md

```diff
@@ -0,0 +1,490 @@
+---
+title: Using Subsquid to Index Your ContainerChain
+description: Learn how to use Subsquid, a query node framework that can index both Substrate and EVM data, to process blockchain data for your Tanssi ContainerChain. 
+---
+
+# Using Subsquid to Index Your ContainerChain
+
+## Introduction {: #introduction }
+
+[Subsquid](https://subsquid.io){target=_blank} is a data network that allows rapid and cost-efficient retrieval of blockchain data from 100+ chains using Subsquid’s decentralized data lake and open-source SDK. In very simple terms, Subsquid can be thought of as an ETL (extract, transform, and load) tool with a [GraphQL](https://graphql.org/){target=_blank} server included. It enables comprehensive filtering, pagination, and even full-text search capabilities.
+
+Subsquid has native and full support for both EVM and Substrate data. Subsquid offers a Substrate Archive and Processor and an EVM Archive and Processor. The Substrate Archive and Processor can be used to index both Substrate and EVM data. This allows developers to extract on-chain data from any of the Moonbeam networks and process EVM logs as well as Substrate entities (events, extrinsics, and storage items) in one single project and serve the resulting data with one single GraphQL endpoint. If you exclusively want to index EVM data, it is recommended to use the EVM Archive and Processor.
+
+This tutorial is a step-by-step guide to building a squid to index EVM data from start to finish. It's recommended that you follow along taking each step described on your own, but you can also find a complete version of Squid built in this tutorial here.  
+
+## Checking Prerequisites {: #checking-prerequisites }
+
+To follow along with this tutorial, you'll need to have:
+
+- [Docker installed](https://docs.docker.com/get-docker/){target=_blank}
+- [Docker Compose installed](https://docs.docker.com/compose/install/){target=_blank}
+- An empty Hardhat project. For step-by-step instructions, please refer to the [Creating a Hardhat Project](/builders/interact/ethereum-api/dev-env/hardhat/#creating-a-hardhat-project){target=_blank} section of our Hardhat documentation page
+
+--8<-- 'text/common/general-js-tutorial-check.md'
+
+## Deploying an ERC-20 with Hardhat {: #deploying-an-erc20-with-hardhat }
```

Comment:
I had another thought I forgot to share in my initial review: This tutorial is quite long; almost half of it is the setup just to have data to index. I feel like most people looking for an indexer will have data to index. Maybe not, though. Maybe we should just provide the user with a contract address and the transfer details for the four transactions (in a table or something) on the demo container.

On the other hand, I also see the value in having this here haha so I don't really know... what do you think?

Reply:
I see value in either option. Because folks might not have much deployed to their ContainerChain yet I think it's important to have those steps there, but there might also be people who want to skip that and move faster. So I included a table with the deployed ERC-20 contract address and hashes for the transfers at the top in the event they are working with the Demo ContainerChain and want to skip the deployment steps.